### PR TITLE
Resolve #805 : Remove Mutex lock

### LINKED ISSUE #805: Remove obsolete locking
**This is an [Easy Hack](https://collaboraonline.github.io/post/easyhacks/).**
Potential mentors: @mmeeks 

### Detailed description and rationale
In recent times we've significantly simplified our threading model in COOL, but there are old / obsolete locking pieces left lying around.

### Code pointers
The kit/Kit.cpp has a _mutex field-  that needs to be removed. While we're there the getMutex() method should go, also from the common/Session base-class

Hopefully an easy one =)

### PR DESCRIPTION
**Pull request by 22shubh22**
Signed-off-by: Shubham Goyal <22shubh22@gmail.com>
Change-Id: I80da9d085590e67c0998ce971225a8c9903c0bb7

* Resolves: # 
* Target version: master 

### Summary
#805 Remove _mutex field

### TODO

- [ ] ...

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: kit/Kit.cpp [hunk 1]
```diff
@@ -606,6 +597,7 @@ class Document final : public DocumentManagerInterface
 #endif
         }
 
+        // TODO: Should I remove the comment?
```
Comment (22shubh22) [inline kit/Kit.cpp:600]: Should I remove the comment?

**Comment by mmeeks** [comment]
Patch is great, needs a real-name =)

File: kit/Kit.cpp (hunk 1 above)
Comment (22shubh22) [reply kit/Kit.cpp:600]: This question still needs an answer.

Comment (mmeeks) [reply kit/Kit.cpp:600]: Oh - yes please do =) there is no deadlock there anymore.

**Comment by 22shubh22** [comment]
@Ashod can it be merged?

File: kit/Kit.cpp [hunk 2]
```diff
@@ -894,7 +876,6 @@ class Document final : public DocumentManagerInterface
 
         std::shared_ptr<ChildSession> session = it->second;
 
-        // Flag and release lock.
         ++_isLoading;
```
Comment (Ashod) [inline kit/Kit.cpp:879]: I imagine `_isLoading` would need to go too. If that's too large a change in this PR, then we can leave it for a later patch (and revert the relevant changes in this function/file).

**Review by Ashod — COMMENTED** [review]
Thanks for the patch. Looks good, but I see partial removal around `_isLoading`. See comments below.

Thanks again for the PR!

File: kit/Kit.cpp (hunk 2 above)
Comment (22shubh22) [reply kit/Kit.cpp:879]: okay, so I will work with isloading on another patch.

Comment (Ashod) [reply kit/Kit.cpp:879]: @22shubh22 good, but you have already made changes that affect `_isLoading`. Best to revert those if you will remove _isLoading in another patch. I fear the partial change might cause some issues.

Shouldn't be difficult to find the related code that I'm referring to, but feel free to ask if you need help.

Thanks again!

**Review by mmeeks — COMMENTED** [review]
Soo -- thanks so much for the change Shubham - to stop this hanging around - I implemented the other changes to remove the _isLoading and pushed to here:
    https://github.com/CollaboraOnline/online/pull/1232

Perhaps it is worth closing this request ?

**Comment by mmeeks** [comment]
Merged your patch as part of that other pull - thank you; any plans on things to do next ? =)

**Pull request closed without merging**